### PR TITLE
Added code for unlimited allowance

### PR DESCRIPTION
**Pull request by rashtrakoff**
Added/modified code in `StreamExchange.sol` and `StreamExchangeHelper.sol` for allowing unlimited allowance (issue #3 ). Test file has been modified for the same by adding 4 expect statements checking for unlimited allowance.

### CODE REVIEW

File: 01-Contracts/contracts/StreamExchange.sol [hunk 1]
```diff
@@ -82,13 +68,31 @@ contract StreamExchange is Ownable, SuperAppBase, UsingTellor {
...
+        // and Supertoken upgrades
+        ERC20(_exchange.inputToken.getUnderlyingToken()).safeIncreaseAllowance(
+            address(_exchange.inputToken),
+            2**256 - 1
+        );
+        ERC20(_exchange.outputToken.getUnderlyingToken()).safeIncreaseAllowance(
+                address(_exchange.outputToken),
+                2**256 - 1
+            );
+
```
Comment (rashtrakoff) [inline 01-Contracts/contracts/StreamExchange.sol:89]: Unlimited allowance code.

File: 01-Contracts/contracts/StreamExchangeHelper.sol [hunk 2]
```diff
@@ -4,362 +4,452 @@ pragma experimental ABIEncoderV2;
...
-    minOutput = minOutput * 1e18 / (10 ** ERC20(outputToken).decimals());
-
-    path = new address[](2);
-    path[0] = inputToken;
-    path[1] = outputToken;
-
-    // Downgrade and scale the input amount
-    self.inputToken.downgrade(amount);
-    // Swap on Sushiswap
-    ERC20(inputToken).safeIncreaseAllowance(address(self.sushiRouter), amount);
```
Comment (rashtrakoff) [inline 01-Contracts/contracts/StreamExchangeHelper.sol:196]: Redundant code. Changes already made in `StreamExchange.sol`

File: 01-Contracts/contracts/StreamExchangeHelper.sol [hunk 3]
```diff
@@ -4,362 +4,452 @@ pragma experimental ABIEncoderV2;
...
-       address(this),
-       deadline
-    );
-    // Assumes `amount` was outputToken.balanceOf(address(this))
-    outputAmount = ERC20(outputToken).balanceOf(address(this));
-    console.log("outputAmount", outputAmount);
-    require(outputAmount >= minOutput, "BAD_EXCHANGE_RATE: Try again later");
-
-    // Convert the outputToken back to its supertoken version
-    ERC20(outputToken).safeIncreaseAllowance(address(self.outputToken), outputAmount);
```
Comment (rashtrakoff) [inline 01-Contracts/contracts/StreamExchangeHelper.sol:210]: Redundant code. Changes already made in StreamExchange.sol

File: 01-Contracts/test/SteamExchange.test.js [hunk 4]
```diff
@@ -10,423 +10,435 @@ const traveler = require("ganache-time-traveler");
...
-      // Admin
-      await hre.network.provider.request({
-        method: "hardhat_impersonateAccount",
-        params: [OWNER_ADDRESS]}
-      )
-      owner = await ethers.provider.getSigner(OWNER_ADDRESS)
-      await deployFramework(errorHandler, {
-          web3,
-          from: owner.address,
-      });
```
Comment (rashtrakoff) [inline 01-Contracts/test/SteamExchange.test.js:67]: This should be removed as we are using Polygon mainnet SF framework. This code is useful for deploying SF test framework only.

File: 01-Contracts/test/SteamExchange.test.js [hunk 5]
```diff
@@ -10,423 +10,435 @@ const traveler = require("ganache-time-traveler");
...
 
-    async function checkOwner() {
-        const owner = await u.admin.address;
-        console.log("Contract Owner: ", aliases[owner], " = ", owner);
-        return owner.toString();
-    }
+    const ERC20 = await ethers.getContractFactory("ERC20");
+    ric = await ERC20.attach(RIC_TOKEN_ADDRESS);
+    weth = await ERC20.attach(await ethx.getUnderlyingToken());
+    usdc = await ERC20.attach(await usdcx.getUnderlyingToken());
```
Comment (rashtrakoff) [inline 01-Contracts/test/SteamExchange.test.js:134]: Necessary lines of code for checking allowance of tokens.

File: 01-Contracts/test/SteamExchange.test.js [hunk 6]
```diff
@@ -10,423 +10,435 @@ const traveler = require("ganache-time-traveler");
...
+      expect(await app.getRequestId()).to.equal(1)
+      expect(await app.getOwner()).to.equal(u.admin.address)
+      expect(await app.getFeeRate()).to.equal(20000)
+
+      // Unlimited allowance checks
+      expect(await weth.allowance(app.address, SUSHISWAP_ROUTER_ADDRESS)).to.be.equal(ethers.constants.MaxUint256);
+      expect(await usdc.allowance(app.address, SUSHISWAP_ROUTER_ADDRESS)).to.be.equal(ethers.constants.MaxUint256);
+      expect(await weth.allowance(app.address, ethx.address)).to.be.equal(ethers.constants.MaxUint256);
+      expect(await usdc.allowance(app.address, usdcx.address)).to.be.equal(ethers.constants.MaxUint256);
+
```
Comment (rashtrakoff) [inline 01-Contracts/test/SteamExchange.test.js:338]: Actual testing code. Checks for unlimited allowance.

**Comment by mikeghen** [comment]
This still has far too much diff, if it isn't broken don't be fixing it, I'm closing this because it's still mostly formating changes, the code you highlighted in your comments should be the only code in a PR to close #17

**Pull request closed without merging**